src/Maestro/Maestro.MergePolicies/VersionDetailsPropsMergePolicy.cs

```diff
@@ -60,14 +70,16 @@ public override async Task<MergePolicyEvaluationResult> EvaluateAsync(PullReques
 
         try
         {
-            var versionProps = DependencyFileManager.GetXmlDocument(await remote.GetFileContentsAsync(
-                VersionFiles.VersionProps,
+            var versionsPropsContent = await remote.GetFileContentsAsync(
+                VersionFiles.VersionsProps,
                 pr.TargetRepoUrl,
-                pr.HeadBranch));
+                pr.HeadBranch);
+            var versionsProps = ProjectRootElement.Create(
+                XmlReader.Create(new StringReader(versionsPropsContent)));
 
-            var versionDetailsPropsProperties = ExtractPropertiesFromXml(versionDetailsProps).Keys.ToHashSet();
+            var versionDetailsPropsProperties = ExtractNonConditionalProperties(versionDetailsProps).Keys.ToHashSet();
             
-            var versionPropsDictionary = ExtractPropertiesFromXml(versionProps);
+            var versionPropsDictionary = ExtractNonConditionalProperties(versionsProps);
```

Comment:
Why does this need to be a dictionary?
What if it extracted non-empty non-conditioned properties?

We'd just have two hash sets and could substract them? The code below could be almost 1 line?

Reply:
true, fair, I think I was printing full lines to remove before and didn't change it